src/Microsoft.Android.Sdk.ILLink/LinkContextExtensions.cs

```diff
@@ -0,0 +1,161 @@
+#nullable enable
+
+using Mono.Cecil;
+using Mono.Collections.Generic;
+using Mono.Linker;
+using System.Collections.Generic;
+
+namespace Microsoft.Android.Sdk.ILLink
+{
+	public static class LinkContextExtensions
+	{
+		public static MethodDefinition GetBaseDefinition (this LinkContext context, MethodDefinition method)
```

Comment:
I find it somewhat "disturbing"/"odd"/"unfortunate" that this would be adding a *different* `GetBaseDefinition()` method in the context of the linker, with a *different* algorithm for comparing method return types (or lack thereof!) and parameter types!  See also:

  * https://github.com/xamarin/java.interop/blob/412e974b9a65c67eab9c5fb5ed330a09abe66710/src/Java.Interop.Tools.Cecil/Java.Interop.Tools.Cecil/MethodDefinitionRocks.cs#L16-L32
  * https://github.com/xamarin/xamarin-android/blob/92efdccb686454be32d742b8ca9f6a166a7c1808/src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/FixAbstractMethodsStep.cs#L109-L158

The method here looks very similar to the one in Java.Interop -- which I now see is *incomplete*, because it doesn't check `MethodDefinition.HasOverrides`.

Which leads one to ask, *how many **different** (and potentially incomplete/wrong) implementations do we have to find the "base" method of an override?!*

I sense a Mono.Cecil feature request…

Reply:
Agreed, in an ideal world the type hierarchy logicwith `Resolve` caching would be built-in to cecil. This version of `GetBaseDefinition` exists just to use a cache that's built-in to the linker in .NET 6. (It's actually copied from the java.interop version). :grimacing:

Reply:
Removed in the latest iteration